spring-kafka/src/main/java/org/springframework/kafka/listener/adapter/MessagingMessageListenerAdapter.java

```diff
@@ -665,16 +671,27 @@ protected void asyncFailure(Object request, @Nullable Acknowledgment acknowledgm
 			Throwable t, Message<?> source) {
 
 		try {
+			Throwable cause = t instanceof CompletionException ? t.getCause() : t;
 			handleException(request, acknowledgment, consumer, source,
-					new ListenerExecutionFailedException(createMessagingErrorMessage(
-							"Async Fail", source.getPayload()), t));
+							new ListenerExecutionFailedException(createMessagingErrorMessage(
+									"Async Fail", source.getPayload()), cause));
 		}
 		catch (Throwable ex) {
 			this.logger.error(t, () -> "Future, Mono, or suspend function was completed with an exception for " + source);
 			acknowledge(acknowledgment);
+			if (canAsyncRetry(request, ex) &&
+				Objects.nonNull(this.callbackForAsyncFailure)) {
+				ConsumerRecord<K, V> record = (ConsumerRecord<K, V>) request;
```

Comment:
```
/home/runner/work/spring-kafka/spring-kafka/spring-kafka/src/main/java/org/springframework/kafka/listener/adapter/MessagingMessageListenerAdapter.java:684: warning: [unchecked] unchecked cast

> Task :spring-kafka:compileJava
				ConsumerRecord<K, V> record = (ConsumerRecord<K, V>) request;
```

Consider to add a `@SuppressWarnings("unchecked")`

Reply:
Github seems to have UI error. 😓 
Fixed it, but Github UI don't be aware of it.
<img width="788" alt="image" src="https://github.com/user-attachments/assets/aa54d47e-a366-4539-bffe-cbbe27802520">

Reply:
Not sure what you'd like to see here.
The `@SuppressWarnings("unchecked")` is in and build is OK.
You might be wondering why this conversation is not closed.
Just because you made a change on a different line.
So, so far so good.
Thanks